### PR TITLE
Added changelog for 6.4.0

### LINKED ISSUE #2306: New Release on version 6?
Hello! what about to launch a new release to version 6 with current commits on master?

Last release is more than one year ago (april 2018) and there are about 100 commits more on master.

Thanks!

### PR DESCRIPTION
**Pull request by Nyholm**
There have been over 115 commits to master since 6.3.3. Most are doc fixes, test fixes and typos. In this PR is a changelog to all significant changes for the code (not tests). 

I think we are ready to release 6.4.0 after this PR. 

This will close #2306

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,17 @@
 # Change Log
 
+## 6.4.0 - 2018-10-23
+
+* Improvement: Improved error messages when using cUrl < 7.21.2 [#2108](https://github.com/guzzle/guzzle/pull/2108)
+* Fix: Test if response is readable before returing a summery in `RequestException::getResponseBodySummary()` [#2081](https://github.com/guzzle/guzzle/pull/2081)
```
Comment (stof) [inline CHANGELOG.md]: typos: `returning` and `summary`

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,17 @@
 # Change Log
 
+## 6.4.0 - 2018-10-23
+
+* Improvement: Improved error messages when using cUrl < 7.21.2 [#2108](https://github.com/guzzle/guzzle/pull/2108)
```
Comment (stof) [inline CHANGELOG.md]: why using a capital U in `cUrl` ? https://curl.haxx.se/ writes it lowercase.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,17 @@
...
+* Fix: Test if response is readable before returing a summery in `RequestException::getResponseBodySummary()` [#2081](https://github.com/guzzle/guzzle/pull/2081)
+* Fix: Add support for GUZZLE_CURL_SELECT_TIMEOUT environment variable [#2161](https://github.com/guzzle/guzzle/pull/2161)
+* Improvement: Added `GuzzleHttp\Exception\InvalidArgumentException` [#2163](https://github.com/guzzle/guzzle/pull/2163)
+* Improvement: Added `GuzzleHttp\_current_time()` to use `hrtime()` if that function exists. [#2242](https://github.com/guzzle/guzzle/pull/2242)
+* Improvement: Added cUrl's `appconnect_time` in `TransferStats` [#2284](https://github.com/guzzle/guzzle/pull/2284)
+* Improvement: Make GuzzleException extend Throwable whereever it's available [#2273](https://github.com/guzzle/guzzle/pull/2273)
```
Comment (stof) [inline CHANGELOG.md]: typo: `wherever`

File: CHANGELOG.md (hunk 2 above)
Comment (Nyholm) [reply CHANGELOG.md]: Oh. I don’t know. I’ve always written it like that. 

Part of me wants to see how it was written 10 years ago :)

I’ll update

Comment (stof) [reply CHANGELOG.md]: according to https://curl.haxx.se/docs/faq.html the original spelling is `cURL`

Comment (Nyholm) [reply CHANGELOG.md]: Thank you

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,17 @@
 # Change Log
 
+## 6.4.0 - 2018-10-23
```
Comment (Tobion) [inline CHANGELOG.md]: wrong year ^^

Comment (Nyholm) [reply CHANGELOG.md]: Thank you =)

Comment (sagikazarmark) [reply CHANGELOG.md]: Nice catch

**Pull request merged**